pkg/apis/apps/validation/validation.go

```diff
@@ -75,6 +75,48 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("podManagementPolicy"), spec.PodManagementPolicy, fmt.Sprintf("must be '%s' or '%s'", apps.OrderedReadyPodManagement, apps.ParallelPodManagement)))
 	}
 
+	switch spec.UpdateStrategy.Type {
+	case "":
+		allErrs = append(allErrs, field.Required(fldPath.Child("updateStrategy"), ""))
+	case apps.OnDeleteStatefulSetStrategyType, apps.RollingUpdateStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition != nil {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					fldPath.Child("updateStrategy").Child("partition"),
+					spec.UpdateStrategy.Partition.Ordinal,
+					fmt.Sprintf("only allowed for updateStrategy '%s'", apps.PartitionStatefulSetStrategyType)))
+		}
+	case apps.PartitionStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition == nil {
```

Comment:
I can get on board with the approach of adding  the partition to rolling update, but I think we can send the condition from the  controller in either case. In order to perform a staged rollout you can use patches like the one below while decrementing the ordinal.
```json
{
	"updateStrategy": {
		"type": "Partitioned",
		"ordinal": <ordinal>
	}
}
```

Since the partitioned strategy never completes, no signal will be sent, via a Condition or otherwise, until user switches the strategy back to rolling update or on delete. So the last applied patch will look like this.

```yaml
{
	"updateStrategy": {
		"type": "RollingUpdate"
	}
}
```

At this point the controller will complete the update and send any signals (including Conditions).

What I like about this approach is that operators, human or otherwise, have an opportunity to validate the deployed product prior to triggering the controller to signal completion. Readiness checks are good for determining if the product is running, but this would allow for adding a smoke test prior completing the update.

Again, I don't feel super strongly about it